integration/test/multireleasehandling/multi_release_handling_test.go

```diff
@@ -170,7 +193,75 @@ func TestMultiReleaseHandling(t *testing.T) {
 
 			return nil
 		}
-		b := backoff.NewMaxRetries(30, 5*time.Second)
+		b := backoff.NewMaxRetries(35, 6*time.Second)
+
+		err := backoff.Retry(o, b)
+		if err != nil {
+			t.Fatalf("err == %v, want %v", err, nil)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checked App CRs created for Release CRs components"))
+	}
+
+	// Create the ReleaseCycle for the Release CR 1 marking it as EOL release.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating ReleaseCycle CR %#q", releaseCycleCR1.Name))
+
+		_, err := config.K8sClients.G8sClient().ReleaseV1alpha1().ReleaseCycles().Create(releaseCycleCR1)
+		if err != nil {
+			t.Fatalf("err == %v, want %v", err, nil)
+		}
+
+		defer func() {
+			if env.CircleCI() || env.KeepResources() {
+				return
+			}
+
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("cleaning up ReleaseCycle CR %#q", releaseCycleCR1.Name))
+
+			err := config.K8sClients.G8sClient().ReleaseV1alpha1().ReleaseCycles().Delete(releaseCycleCR1.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				config.Logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("failed to clean up ReleaseCycle CR %#q", releaseCycleCR1.Name), "stack", fmt.Sprintf("%#v", err))
+			}
+
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("cleaned up ReleaseCycle CR %#q", releaseCycleCR1.Name))
+		}()
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created ReleaseCycle CR %#q", releaseCycleCR1.Name))
+	}
+
+	// Check if App CRs for all components of Release CR 1 not shared with
+	// the Release CR 2 are deleted.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking App CRs created for Release CRs components"))
+
+		o := func() error {
+			list, err := config.K8sClients.G8sClient().ApplicationV1alpha1().Apps("").List(metav1.ListOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			var appCRNames []string
+			for _, obj := range list.Items {
+				appCRNames = append(appCRNames, obj.Name)
+			}
+
+			expectedAppCRNames := []string{
+				"aws-operator.4.7.0",
+				"cert-operator.0.1.0",
+				"release-aws.v6.1.0",
+			}
+
+			sort.Strings(appCRNames)
+			sort.Strings(expectedAppCRNames)
+
+			if !cmp.Equal(appCRNames, expectedAppCRNames) {
+				return microerror.Maskf(waitError, "\n\n%s\nappCRNames = %#v\nexpectedAppCRNames = %#v\n\n", cmp.Diff(appCRNames, expectedAppCRNames), appCRNames, expectedAppCRNames)
```

Comment:
The cmp.Diff output is not sufficient for me when it comes to slices:

```
{[]string}[?->0]:
        -: <non-existent>
        +: "aws-operator.4.6.0"
```

I extended it so it prints something like:

```
{[]string}[?->0]:
        -: <non-existent>
        +: "aws-operator.4.6.

appCRNames = []string{"aws-operator.4.7.0", "cert-operator.0.1.0"}
expectedAppCRNames = []string{"aws-operator.4.6.0", "aws-operator.4.7.0", "cert-operator.0.1.0"}
```

I'm thinking about creating our own `cmp` library that would use `github.com/google/go-cmp/cmp` under the hood but with more verbose output.

Something like `cmp.Diff(appCRNames, expectedAppCRNames)`. Then it would output:

```
{[]string}[?->0]:
        -: <non-existent>
        +: "aws-operator.4.6.

left = []string{"aws-operator.4.7.0", "cert-operator.0.1.0"}
right = []string{"aws-operator.4.6.0", "aws-operator.4.7.0", "cert-operator.0.1.0"}
```

Reply:
diff look ok to me, I don't see a need to improves this

Reply:
Lucky you. I'm freaking out when I see this.